packages/pwa-kit-runtime/src/utils/ssr-server.test.js

```diff
@@ -960,41 +959,6 @@ describe('updateGlobalAgentOptions', () => {
     })
 })
 
-describe('PerformanceTimer tests', () => {
-    let timer
-    beforeEach(() => (timer = new PerformanceTimer('')))
-    afterEach(() => timer?._observer.disconnect())
-
-    test('time() function', () => {
-        const func = sinon.stub()
-        timer.time('test1', func, 1)
-        expect(func.callCount).toBe(1)
-        expect(func.calledWith(1)).toBe(true)
-
-        func.reset()
-        func.throws('Error', 'intentional error')
-        expect(() => timer.time('test2', func, 1)).toThrow('intentional error')
-        expect(func.callCount).toBe(1)
-    })
-
-    test('start() and end()', async () => {
-        timer.start('test3')
-        await new Promise((resolve) => setTimeout(resolve, 10))
-        timer.end('test3')
-        const summary = timer.summary
-        expect(summary.length).toBe(1)
-        const entry = summary[0]
-        expect(entry.name).toEqual('test3')
-        expect(entry.duration).toBeGreaterThanOrEqual(5)
-        expect(entry.duration).toBeLessThanOrEqual(30)
-    })
-
-    test('accessing operationId auto-increments the counter', () => {
-        expect(timer.operationId).toBe(1)
-        expect(timer.operationId).toBe(2)
-    })
-})
-
```

Comment:
@raiyaj it looks like most of this PR thus far is about dropping `PerformanceTimer`, is it not compatible with node 16? Is there a node 16 built-in API that makes it unnecessary? I suppose if "yes", I can understand the reasoning behind not unit testing the node 16 `timer` API if that's the situation

Reply:
Thanks for taking a look @bfeister! Yup, our custom `PerformanceTimer` isn't compatible with node 16 as it it - I get this when I run `npm start` in a newly-generated project:
```
SyntaxError: The "reqbb0e1f8e-ea1f-4a84-afc5-c54aba83056d-ssr-overall-start" performance mark has not been set
    at new DOMException (node:internal/per_context/domexception:72:5)
    at __node_internal_ (node:internal/util:497:10)
    at getMark (node:internal/perf/usertiming:58:11)
    at calculateStartDuration (node:internal/perf/usertiming:141:13)
    at EventTarget.measure (node:internal/perf/usertiming:159:7)
    at PerformanceTimer.end (/Users/rjessa/generated-test-project/node_modules/pwa-kit-runtime/utils/ssr-server/performance-timer.js:117:29)
    at ServerResponse.done (/Users/rjessa/generated-test-project/node_modules/pwa-kit-runtime/ssr/server/build-remote-server.js:934:11)
    at ServerResponse.emit (node:events:525:35)
    at Socket.onServerResponseClose (node:_http_server:258:23)
    at Socket.emit (node:events:525:35)
    at TCP.<anonymous> (node:net:757:14)
error: Command failed: node /Users/rjessa/generated-test-project/app/ssr.js
```

The [comment](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/pwa-kit-runtime/src/utils/ssr-server/performance-timer.js#L10-L12) in the `PerformanceTimer` class says it wraps the node performance measurement API ([`perf-hooks`](https://nodejs.org/docs/latest-v18.x/api/perf_hooks.html)) and was created to guard against changes to it, back when the builtin lib was at Stability 1. Now that it's at Stability 2 for node 14/16/18, we were hoping to remove `PerformanceTimer` rather than updating our usage of `perf-hooks`. But I suppose this is a breaking change if anyone is using `PerformanceTimer` or any of the "marks" we've added like `ssr-overall`, `cache-response`, etc.

Our ultimate goal for this PR is to add explicit support for node 16 _and_ 18. So far it seems like these changes makes that possible - the only other issue I've found is some transitive dependencies don't seem to have explicit support for node 18. I get this when I run `npm ci` in my project:
```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'retail-react-app@0.0.1',
npm WARN EBADENGINE   required: { node: '^14.0.0', npm: '^6.14.4 || ^7.0.0 || ^8.0.0' },
npm WARN EBADENGINE   current: { node: 'v18.14.0', npm: '9.3.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'eslint-import-resolver-webpack@0.13.1',
npm WARN EBADENGINE   required: {
npm WARN EBADENGINE     node: '^16 || ^15 || ^14 || ^13 || ^12 || ^11 || ^10 || ^9 || ^8 || ^7 || ^6'
npm WARN EBADENGINE   },
npm WARN EBADENGINE   current: { node: 'v18.14.0', npm: '9.3.1' }
npm WARN EBADENGINE }
```

Let me know what you think!

Reply:
On second thought, I'm going to take a stab at fixing `PeformanceTimer` for node 16, rather than removing it, to avoid making a breaking change. And it turns out the MRT platform actually doesn't support node 18 yet, so I'm going to focus on node 16 support only.